### PR TITLE
Pin the version of OLM we use for e2e tests

### PR DESCRIPTION
**Pull request by dgn**
The latest release does not work, but 0.28.0 does (see https://github.com/operator-framework/operator-lifecycle-manager/issues/3419). I also added some automation to update-deps to update the version.

### CODE REVIEW

**Review by bmangoen — DISMISSED** [review]

File: tests/e2e/common-operator-integ-suite.sh [hunk 1]
```diff
@@ -241,9 +241,12 @@ if [ "${SKIP_BUILD}" == "false" ]; then
...
     OPENSHIFT_PLATFORM=false \
     make bundle bundle-build bundle-push
-     
+
     # Install OLM in the cluster because it's not available by default in kind.
-    ${OPERATOR_SDK} olm install
+    if [ "${OLM_VERSION}" != "" ]; then
+      OLM_INSTALL_ARGS="--version ${OLM_VERSION}"
+    fi
+    ${OPERATOR_SDK} olm install "${OLM_INSTALL_ARGS}"
```
Comment (FilipB) [inline tests/e2e/common-operator-integ-suite.sh:249]: Isn't this failing when OLM_INSTALL_ARGS is undefined?

File: tools/update_deps.sh [hunk 2]
```diff
@@ -58,6 +58,10 @@ sed -i "s|CONTROLLER_TOOLS_VERSION ?= .*|CONTROLLER_TOOLS_VERSION ?= ${CONTROLLE
 OPM_LATEST_VERSION=$(getLatestVersion operator-framework/operator-registry)
 sed -i "s|OPM_VERSION ?= .*|OPM_VERSION ?= ${OPM_LATEST_VERSION}|" "${ROOTDIR}/Makefile.core.mk"
 
+# Update olm
+OLM_LATEST_VERSION=$(getLatestVersion operator-framework/operator-registry)
```
Comment (bmangoen) [inline tools/update_deps.sh:62]: Typo here `operator-framework/operator-lifecycle-manager`

File: tests/e2e/common-operator-integ-suite.sh (hunk 1 above)
Comment (FilipB) [reply tests/e2e/common-operator-integ-suite.sh:249]: It will fail if undefined as the script is using `set -eu -o pipefail`

Comment (dgn) [reply tests/e2e/common-operator-integ-suite.sh:250]: good point. fixed

File: tools/update_deps.sh (hunk 2 above)
Comment (dgn) [reply tools/update_deps.sh]: Thanks, fixed!

**Review by FilipB — APPROVED** [review]
Thank you

**Review by jwendell — APPROVED** [review]

**Pull request merged**